GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -1324,7 +1325,17 @@ void OnRevisionReadCompleted()
                         }
                         else
                         {
-                            parents = TryGetParents(Module, _filterInfo, notSelectedId);
+                            if (notSelectedId.IsArtificial)
+                            {
+                                notSelectedId = CurrentCheckout is not null
+                                    ? CurrentCheckout
+                                    : _gridView.ToBeSelectedObjectIds.Where(id => !id.IsArtificial).FirstOrDefault();
+                            }
+
+                            if (notSelectedId is not null)
+                            {
+                                parents = TryGetParents(Module, _filterInfo, notSelectedId);
+                            }
```

Comment:
To keep it simple, the following can be reduced with the drawback that unselecting ShowArtificial while artificial is selected will not select anything

```suggestion
                            if (!notSelectedId.IsArtificial)
                            {
                                parents = TryGetParents(Module, _filterInfo, notSelectedId);
                            }
```

Reply:
Simplified this, added a comment instead
(The snippet was written when the solution was slightly different, I kept it though)